javaCreateAccountCore/src/main/java/com/hamdan/forzenbook/java/createaccount/core/viewmodel/JavaCreateAccountViewModel.java

```diff
@@ -0,0 +1,207 @@
+
+    @Inject
+    public JavaCreateAccountViewModel(CreateAccountValidationUseCase createAccountValidationUseCase, CreateAccountUseCase createAccountUseCase) {
+        this.createAccountValidationUseCase = createAccountValidationUseCase;
+        this.createAccountUseCase = createAccountUseCase;
+    }
+
+
+    public Flowable<JavaCreateAccountState> getState() {
+        // behaviorSubject.getValue()
+        return _state.toFlowable(BackpressureStrategy.BUFFER);
+    }
+
+    private JavaCreateAccountState viewableState() {
+        return _state.getValue();
+    }
+
+    private void setState(JavaCreateAccountState inState) {
+        _state.onNext(inState);
+    }
+
+    public void createAccountDateDialogClicked() {
+        createAccountShowDateDialog();
+    }
+
+    public void createAccountDateDialogSubmitClicked() {
+        closeAccountShowDateDialog();
+    }
+
+    public void createAccountDateDialogDismiss() {
+        closeAccountShowDateDialog();
+    }
+
+    public void createAccountDismissErrorClicked() {
+        createAccountNormalView();
+    }
+
+    private void closeAccountShowDateDialog() {
+        setState(new JavaCreateAccountState(
+                viewableState().getErrorId(),
+                viewableState().getFirstName(),
+                viewableState().getLastName(),
+                viewableState().getBirthDay(),
+                viewableState().getLocation(),
+                viewableState().getEmail(),
+                false,
+                viewableState().getLoading()
+        ));
+    }
+
+    private void createAccountShowDateDialog() {
+        setState(
+                new JavaCreateAccountState(
+                        viewableState().getErrorId(),
+                        viewableState().getFirstName(),
+                        viewableState().getLastName(),
+                        viewableState().getBirthDay(),
+                        viewableState().getLocation(),
+                        viewableState().getEmail(),
+                        true,
+                        viewableState().getLoading()
+                )
+        );
+    }
+
+    private void createAccountNormalView() {
+        setState(
+                new JavaCreateAccountState(
+                        null,
+                        viewableState().getFirstName(),
+                        viewableState().getLastName(),
+                        viewableState().getBirthDay(),
+                        viewableState().getLocation(),
+                        viewableState().getEmail(),
+                        viewableState().getDateDialogOpen(),
+                        viewableState().getLoading()
+                )
+        );
+    }
+
+    public void updateCreateAccountTextAndErrors(Entry firstName, Entry lastName, Entry birthDay, Entry email, Entry location) {
+        CreateAccountEntrys stringStates = createAccountValidationUseCase.invoke(toCreateAccountEntrys(firstName, lastName, birthDay, email, location));
```

Comment:
Ugly formatting, reformat please

Reply:
I think it's better now? Will need you to look at it again ofc